### PR TITLE
Drop EoL Debian, adjust deps for Puppet 6 & 7

### PR DESCRIPTION
**Pull request by genebean**
#### Pull Request (PR) description

Module dependencies were updated minimum versions that support Puppet 6 and max versions that includes the latest releases.

The "description" key was removed as it is no longer supported.

yumrepo_core was also dropped as it is only needed in the fixtures file.

yum was dropped because it is not used within this module

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -7,19 +7,18 @@
...
     },
     {
       "name": "puppetlabs/apt",
-      "version_requirement": ">= 2.0.0 < 8.0.0"
+      "version_requirement": ">= 6.1.0 < 9.0.0"
     },
     {
       "name": "puppet/yum",
-      "version_requirement": ">= 0.9.6 < 5.0.0"
+      "version_requirement": ">= 3.1.1 < 5.0.0"
```
Comment (ekohl) [inline metadata.json:21]: Why does it depend on `puppet/yum`? I don't see it used anywhere.

File: metadata.json [hunk 2]
```diff
@@ -7,19 +7,18 @@
...
     },
     {
       "name": "puppetlabs/yumrepo_core",
```
Comment (ekohl) [inline metadata.json:24]: I think typically `yumrepo_core` is not listed as a dependency since it's bundled in the regular puppet-agent package.

Comment (genebean) [reply metadata.json:24]: Agreed. I can drop it from here and just make sure it is in the fixtures file instead.

File: metadata.json (hunk 1 above)
Comment (genebean) [reply metadata.json]: I shall drop it too :)

Comment (genebean) [reply metadata.json]: dropped

File: metadata.json (hunk 2 above)
Comment (genebean) [reply metadata.json]: dropped

**Comment by bastelfreak** [comment]
why do you drop debian 9? that is not yet EOL. doesnt it work anymore?

**Comment by attachmentgenie** [comment]
ubuntu 16.04 should be dropped for the same reason https://endoflife.software/operating-systems/linux/ubuntu

**Comment by attachmentgenie** [comment]
Screenshot from 2021-05-10 21-57-30

for me its showing EOL for 16.04

**Comment by genebean** [comment]
@attachmentgenie You misunderstand - this module only lists the newer two:
https://github.com/voxpupuli/puppet-hashi_stack/blob/1844a30bf91b7d83d78fc7914789dc5360d1369f/metadata.json#L41-L47

**Comment by attachmentgenie** [comment]
ah yes, must be going crazy then. ah wait its the consul module that still supports it.

**Review by attachmentgenie — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

**Pull request merged**